New Features 1

sbridger opened the pull request:
Preliminary pull request, to establish that I can push working changes all the way back to you. 
QML is new to me, still trying to get my head around binding loops and so on, and thus there may be *very bad code* 

- I have yet to merge gpsinfo.pro successfully.

Features:
- Add TimeToFirstFix  to aid establishing how long it takes to get lock.
- added declination, magnetic north line and movement direction lines to help checking compass operation (TLDR: it's crap)
- SatInfo: sequencing+transparency so that  overlapping numbers are mostly legible now.
- barchart now shows active sats in color, inactive in mono. https://github.com/direc85/harbour-gpsinfo/issues/16
   (Still contains code for using borders, but it does nothing...)

sbridger committed:
- SatelliteBar now shows active sats in color.
- SatInfo
  Add magnetic north line
  Add Heading line when moving
  Put active sats over inactive ones
  use transparency to see overlaying numbers
- Add Declination setting
- show magnetic variation
- improve lastUpdate time
- add TimeToFirstFix

sbridger committed:
comment out sharepage for now, trying to fix binding loops

sbridger committed:
Qchart makes unknown component M300 error disappear

sbridger committed:
playing with the grid labels

sbridger committed:
trying to fix binding loops

sbridger committed:
removed

sbridger committed:
Merge branch 'HEAD' of https://github.com/sbridger/harbour-gpsinfo.git

# Conflicts:
#	harbour-gpsinfo.pro

sbridger committed:
ver ->13.3-1

direc85 commented:
Thanks! Sounds good! I'll have to look at this later this week, so you have still time to refine your work, if you want ;)

direc85 commented on qml/pages/FirstPage.qml:
```diff
@@ -23,6 +24,46 @@ Page {
+            property bool pendingFix : true
+            property int secsSincePosition : Math.round((new Date() - positionSource.position.timestamp)/1000)
+
+            property int secs2FF : 0
+            property string secs2FFStr: roundElapsedTime(secs2FF)
+            property string tTFF: {
```
I don't have a clue what "FF" means in this context (maybe it becomes apparent later on?) so I may rename them before merging.

sbridger replied:
FirstFix.  I will rename.
I am moving this timing into providers.qml, so it is visible to cover.qml also.

direc85 replied:
This was moved to `Providers.qml`, and I renamed (some of?) the shortened variables.

direc85 commented on qml/pages/FirstPage.qml:
```diff
@@ -23,6 +24,46 @@ Page {
+                }
+                return secs2FFStr
+            }
+            function t2secs2FF(T){
+                secs2FF = Math.round((T - activateGPSTimestamp)/1000)
+            }
```
One line functions are usually not worth it. This is used only once (twice) so I think it can be inlined.

direc85 replied:
This seems to have been removed/inlined.

direc85 commented on qml/pages/FirstPage.qml:
```diff
@@ -79,13 +122,18 @@ Page {
                     }
                 }
             }
+//            MenuItem {
+//                enabled: gpsDataSource.active
+//                text: qsTr("Share Location")
+//                onClicked: pageStack.push(Qt.resolvedUrl("SharePage.qml"))
+//            }
```
Can you please push SharePage.qml too (in its current state), although it's commented out?

sbridger replied:
I think I will remove it, unless I make some progress on it this week.  
I thought I have a working shargepage.qml last year, but seems to have gone.

direc85 replied:
Removed in commit 2a4b2cd

direc85 commented on qml/pages/FirstPage.qml:
```diff
@@ -23,6 +24,46 @@ Page {
+              firstFixTimestamp = activateGPSTimestamp = new Date()
+              pendingFix =true
+            }
+            function roundElapsedTime(T) {
+                if (T<=90) return Math.round(T)+ "sec"
+                T=T/60;
+                if (T<=90) return LocationFormater.roundToDecimal(T,1)+ "min"
+                T=T/60;
+                return LocationFormater.roundToDecimal(T,1)+ "hr"
+            }
```
I think this should be "formatElapsedTime(T). Perhaps this could be a global function instead? It could be used elsewhere later, too.

sbridger replied:
yes. I will move to a lib

direc85 commented on qml/pages/SatelliteInfoPage.qml:
```diff
@@ -114,11 +116,62 @@ Page {
+        // first draw them all solid , hiding the radar chart background, and ensuring colors are correct
         Repeater {
-            model: status === PageStatus.Inactive ? [] : gpsDataSource.satellites
+            model: status === PageStatus.Inactive ? [] : gpsDataSource.satellites.sort(function(a,b) {return (a.inUse ? 1:-1) - (b.inUse ? 1:-1)})
```
Perhaps this sorting could be done in gpsDataSource instead?

sbridger replied:
Can't do that at the QML level (can't ad/change fields in providers.qml), but I have removed the duplicate in satinfo now.
Not sure if it makes sense in the c++, since it's specific to the satinfo page

direc85 replied:
Yeah, there's no sorting `QMap`, so it isn't even possible in C++ side.

sbridger committed:
remove duplicate satellite sort

sbridger committed:
Added highlight property for value

sbridger committed:
- Moved menu from firstpage to MainMenu, and added to all 3 pages
- Timing now in Providers.qml
- a timer added to keep times displaying when gps is not updating

sbridger commented:
OK, so now it's a lot closer. I have pushed the changes, before I update qdevelop.

Haven't worked out how to get "timing" into firstpage.qml without the longform "providers.timing.secsSincePosition"

I will work on it some more this week, and tidy these issues up
If you could just check that you can get and use my branch OK

direc85 committed:
Merge branch 'master' into master

direc85 committed:
Set satellites in use for emulator

direc85 committed:
Use iterators instead of foreach

direc85 committed:
Rename variables and functions

direc85 committed:
Disable initial animation

direc85 committed:
Fix borders in QChart

This need updated QChart in order to work; that work will be done later.

direc85 committed:
Update QChart (per bar color, border etc.)

direc85 committed:
Remove share location menu item
